test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -14040,6 +14040,9 @@ def enable_cgroup_cset(self):
                  'input-file': path}
             self.server.manager(MGR_CMD_IMPORT, HOOK, a,
                                 'pbs_cgroups')
+            self.log_match('pbs_cgroups.CF;copy hook-related '
+                           'file request received',
+                           starttime=self.server.ctime)
```

Comment:
I've tried this on our local UV-like system and this is failing:

020-05-12 09:33:37,164 INFOCLI  uv-02-s12: /opt/pbs/bin/pbsnodes -s uv-02-s12.pbspro.com -v uv-02-s12
2020-05-12 09:33:37,245 ERROR    err: ['/opt/pbs/bin/pbsnodes: Server has no node list']
2020-05-12 09:33:37,246 INFOCLI2 uv-02-s12: sudo -H /opt/pbs/sbin/pbs_mom --version
2020-05-12 09:33:37,288 INFOCLI2 uv-02-s12: sudo -H /opt/tools/wrappers/rm -f /var/spool/pbs/mom_priv/epilogue
2020-05-12 09:33:37,306 INFOCLI2 uv-02-s12: sudo -H /opt/tools/wrappers/rm -f /var/spool/pbs/mom_priv/prologue
2020-05-12 09:33:37,322 INFOCLI  uv-02-s12: sudo -H /opt/pbs/sbin/pbs_mom -s list
2020-05-12 09:33:37,385 INFOCLI2 uv-02-s12: sudo -H /opt/tools/wrappers/cp /tmp/PtlPbsg6srmjew /var/spool/pbs/mom_priv/config
2020-05-12 09:33:37,404 INFOCLI2 uv-02-s12: /opt/tools/wrappers/cat /proc/mounts
2020-05-12 09:33:37,415 INFO     manager on uv-02-s12: import hook pbs_cgroups {'content-type': 'application/x-config', 'content-encoding': 'default', 'input-file': '/tmp/cfgztxy6wet.json'}
2020-05-12 09:33:37,416 INFOCLI  uv-02-s12: sudo -H /opt/pbs/bin/qmgr -c import hook pbs_cgroups application/x-config default /tmp/cfgztxy6wet.json
2020-05-12 09:33:37,506 ERROR    err: ["hook 'pbs_cgroups' contents overwritten"]
2020-05-12 09:33:37,517 INFO     mom uv-02-s12 log match: searching for "pbs_cgroups.CF;copy hook-related file request received" - with existence - No match
2020-05-12 09:33:38,043 INFO     mom uv-02-s12 log match: searching for "pbs_cgroups.CF;copy hook-related file request received" - with existence - attempt 2
2020-05-12 09:33:38,566 INFO     mom uv-02-s12 log match: searching for "pbs_cgroups.CF;copy hook-related file request received" - with existence - attempt 3
2020-05-12 09:33:39,093 INFO     mom uv-02-s12 log match: searching for "pbs_cgroups.CF;copy hook-related file request received" - with existence - attempt 4
2020-05-12 09:33:39,619 INFO     mom uv-02-s12 log match: searching for "pbs_cgroups.CF;copy hook-related file request received" - with existence - attempt 5
..
2020-05-12 09:34:07,952 INFO     mom uv-02-s12 log match: searching for "pbs_cgroups.CF;copy hook-related file request received" - with existence - attempt 59
2020-05-12 09:34:08,478 INFO     mom uv-02-s12 log match: searching for "pbs_cgroups.CF;copy hook-related file request received" - with existence - attempt 60
2020-05-12 09:34:08,992 INFO     FAILED

----------
Problem is this:
qmgr -c "import hook application/x-config default pbs_cgroups.json"   <-- problem here since there's no local node, server would not communicate the pbs_cgroups.CF file yet to the mom.
restart pbs_mom
check for "pbs_cgroups.CF; copy request"   but since there's no local node yet, then we would never get this message.
---
I'm working on an adjustment to your fix, where after the delete node @default, need to do create node <local_node>, and then do the enable_cgroups_cpuset(), and finally restarting mom.

Reply:
Yes. Fixed since.

Since the race seemed to be that when you create the node (when the hook is disabled) and then immediately enable the node, you have two conflicting HK files being pushed to the node after the server hello, I changed to reconfigure the hook while the nodes are deleted. But then you need to skip those tests for the hook file copies -- they will occur when the node is created later.

You DO NOT want to create the nodes. There was one, and it was removed ON PURPOSE, because that was what created a race condition between the "create node" and the "qmgr set hook X enabled = true".

But you do want to skip the hook copies that will not occur if the nodes are not there, of course.

Reply:
Ah -- I see I forgot about the one in that function, yes.

Reply:
Let me fix it.

Reply:
I think we should simply be guarding this after having done a status() on the node. If it's there, then look for the hook copy, if not, then skipt it -- the hook copy is then done when the node is created.

Reply:
Something like:
            try:
                statm = self.status(NODE, id=self.shortname)
            except:
                statm = []
            if len(statm) >= 1:

Reply:
Please look at the latest push.

Reply:
Latest push resolved our differences :-)